test/service/test_manifest.py

```diff
@@ -503,7 +509,9 @@ def test_terra_bdbag_manifest(self):
 
     @mock_sts
     @mock_s3
-    def test_full_metadata(self):
+    @mock.patch('azul.service.manifest_service.ManifestService._can_use_cached_manifest')
```

Comment:
I'm a little doubtful about the blanket patch of `_can_use_cached_manifest`. Some tests, like this one, pass without this patch. You should be more deliberate in your choice in which tests need patching. If multiple tests need the same patch, you should write a custom patch decorator and explain the reason for the patch in a comment or docstring. If I remove this patch from all methods, the ones that fail all fail in the same way: a KeyError due to a missing `Expiration` key in moto's mock `HEAD` response.  A more surgical patch might therefore be one that targets `StorageService.head` to return include the `Expiration` key. The advantage of a more surgical patch is that more code is covered by unit tests.